modules/mono/glue/GodotSharp/GodotSharp/Core/GodotObject.base.cs

```diff
@@ -38,6 +38,9 @@ bool refCounted
         {
             if (NativePtr == IntPtr.Zero)
             {
+                if (nativeCtor == null)
+                    throw new NativeConstructorNotFoundException(nativeName);
```

Comment:
Add null checking for `nativeCtor`. However, those non-instantiable objects are created from the engine and have `NativePtr` already, so this would never be thrown.

Reply:
I'm not sure if I like this check, specially if it's never going to be true. What do you think about using `Debug.Assert` instead?

Reply:
I don't like it either :>